src/Device/Driver/Larus.cpp

```diff
@@ -0,0 +1,481 @@
+  // Currently not supported by XCSoar
+  //case 'I':
+  //  info.ProvideExternalInstantaneousWind(wind); 
+  //  break;
+  default:
+    return false;
+  }
+
+  return true;
+}
+
+//-----------------------------------------------------------------------------
+/*
+$PLARS Settings parameters bidirectional
+
+       1 2 3   4
+       | | |   |
+$PLARS,a,a,xxx*hh<CR><LF>
+
+Examples:
+$PLARS,L,MC,1.3*1E
+$PLARS,L,BAL,1.25*6B
+$PLARS,L,BUGS,15*3B
+$PLARS,L,QNH,1013.2*74
+
+$PLARS,H,MC,2.1*1B
+$PLARS,H,BAL,1.00*68
+$PLARS,H,BUGS,0*0B
+$PLARS,H,QNH,1031.4*76
+
+The $PLARS record is intended for exchanging setting values between Larus and a host system such as XCSoar. The record can be used in both directions: from host to Larus or from Larus to host.
+
+These records should not be sent cyclically, but only when needed during initialization and when changes are made.
+
+    Data source (L: Larus, H: Host)
+    Settings parameter
+        MC MacCready (0.0 - 9.9)
+        BAL Ballast (1.00 - 1.60)
+        BUGS Bugs in % (0 - 30)
+        QNH QNH in hPa
+    Value (format depends on settings parameter, see examples)
+    Checksum
+*/
+bool
+LarusDevice::PLARS(NMEAInputLine &line, NMEAInfo &info)
+{ // the line starts with the host indicator
+  if (line.ReadOneChar() == 'L') {
+    double value;
+    auto field = line.ReadView();
+    if (line.ReadChecked(value)) {
+      if (field == "MC"sv) {
+        // - value is MacCReady in m/s [0.0 .. 9.9]
+        return info.settings.ProvideMacCready(value, info.clock);
+      } else if (field == "BUGS"sv) {
+        // - value is bugs in % [0-30], OpenSoar wants [0.5 .. 1.00]
+        return info.settings.ProvideBugs(1.0 - value/100.0, info.clock);
+      } else if (field == "BAL"sv) {
+        // - value is ballast overload [1.00..1.60]
+        return info.settings.ProvideBallastOverload(value, info.clock);
+      } else if (field == "QNH"sv) {
+        // - value is pressure in hPa
+        return info.settings.ProvideQNH(AtmosphericPressure::HectoPascal(value),
+                                        info.clock);
+      }
+    }
+  }
+  return false;
+}
+
+//-----------------------------------------------------------------------------
+bool
+LarusDevice::PutBugs(double bugs, OperationEnvironment &env)
+{
+  // $PLARS,H,BUGS,0*0B
+  // OpenSoar/XCSoar has values between [0.5 ..1.0]
+  if ((bugs < 0.7) || (bugs > 1.0))
+    return false;
+  return SendCmd("BUGS,%0.0f", (1.0-bugs) * 100.0, env);
+}
+
+//-----------------------------------------------------------------------------
+bool
+LarusDevice::PutMacCready(double mc, OperationEnvironment &env)
+{
+  // $PLARS,H,MC,2.1*1B
+  return SendCmd("MC,%0.1f", mc, env);
+}
+
+//-----------------------------------------------------------------------------
+bool
+LarusDevice::PutBallast([[maybe_unused]] double fraction, double overload, 
+                        OperationEnvironment &env)
+{ 
+  // $PLARS,H,BAL,1.00*68
+  if ((overload < 1.0) || (overload > 1.60))
```

Comment:
overload, can be less than 1.

Reply:
Good advice -> switch to fraction

Reply:
Out of curiosity, how is this value used by laurus?
I don't see anyway of defining a polar, empty weight or pilot weight, where this could be applied to?
Is this functionality for a display?

Reply:
Yes, that's right. The Larus sensor box can do little with this. We are working on a display. It is not public at the moment, but we will also release it under an open source licence. Bidirectional coupling with XCSoar is important for this.